Tests fails with new version of ibm-db-sa package

When we let `ibm-db-sa==0.4.2` install tests fail with:
```
$ python3 -m pytest tests/system/data_sources/test_db2.py -k column_validation_core_types
...
E                   sqlalchemy.exc.DBAPIError: (ibm_db_dbi.Error) ibm_db_dbi::Error: [IBM][CLI Driver][DB2/LINUXX8664] SQL0413N  Overflow occurred during numeric data type conversion.  SQLSTATE=22003 SQLCODE=-413
E                   (Background on this error at: https://sqlalche.me/e/20/dbapi)
env/lib/python3.12/site-packages/ibm_db_dbi.py:1792: DBAPIError
...
FAILED tests/system/data_sources/test_db2.py::test_column_validation_core_types - sqlalchemy.exc.DBAPIError: (ibm_db_dbi.Error) ibm_db_dbi::Error: [IBM][CLI Driver][DB2/LINUXX8664] SQL0413N  Overflow occurred during numeric data type conversion.  SQLSTATE=220...
FAILED tests/system/data_sources/test_db2.py::test_column_validation_core_types_to_bigquery - sqlalchemy.exc.DBAPIError: (ibm_db_dbi.Error) ibm_db_dbi::Error: [IBM][CLI Driver][DB2/LINUXX8664] SQL0413N  Overflow occurred during numeric data type conversion.  SQLSTATE=220...
```

We don't install this as a standard package so there is no entry in `setup.py` to modify.

For now we've capped the version we use for testing to `0.4.1` until we have time to investigate.

--- a/noxfile.py
+++ b/noxfile.py
@@ -309,9 +309,8 @@ def integration_db2(session):
     """Run DB2 integration tests.
     Ensure DB2 validation is running as expected.
     """
-    _setup_session_requirements(
-        session, extra_packages=["ibm-db-sa", "ibm-db<3.2.7"]
-    )
+    # TODO Remove dependency "ibm-db<3.2.7" below when working on issue-1591.
+    _setup_session_requirements(session, extra_packages=["ibm-db-sa", "ibm-db<3.2.7"])
 
     expected_env_vars = [
         "PROJECT_ID",

--- a/noxfile.py
+++ b/noxfile.py
@@ -309,9 +309,8 @@ def integration_db2(session):
     """Run DB2 integration tests.
     Ensure DB2 validation is running as expected.
     """
-    # TODO Remove dependency "ibm-db<3.2.7" below when working on issue-1591.
     _setup_session_requirements(
-        session, extra_packages=["ibm-db-sa<0.4.2", "ibm-db<3.2.7"]
+        session, extra_packages=["ibm-db-sa", "ibm-db<3.2.7"]
     )
 
     expected_env_vars = [

